platform/engineering/design-docs/2020-06-02-Sentry10.md

```diff
@@ -0,0 +1,142 @@
+
+### Privacy Concerns
+No additional privacy concerns to the ones we have with our existing Sentry.
+
+### Open Questions and Risks
+#### 1. Performance Scaling
+Since we intend on deploying to a single EC2 instance, there is not much room for scaling up in the event that Sentry 10 runs into CPU or memory bottlenecks.
+
+#### 2. Moving Away from `getsentry/onpremise` Happy Path
+The [`getsentry/onpremise` repo](https://github.com/getsentry/onpremise) is a great way to quickly run your own "on premise" Sentry 10 application. It was not designed with scale in mind. In this design, we will be heavily relying on this bootstrapping repo for building and deploying. It's likely that we will eventually have a need to do something custom that would require us to fork the `getsentry/onpremise` repo, making any future modifications and upgrades difficult.
+
+### Work Estimates
+#### Phase 1 - Ansible Build
+Build the AWS AMI. In this step, all docker images should be downloaded onto the machine. This would be considered the ansible "build".
+
+#### Phase 2a - Ansible Deployment
+Create the ansible code to deploy the Sentr10 AMI. This might just be a deployment configuration or it might require new deployment ansible roles or modification to existing roles.
+
+#### Phase 2b - AWS Deplyoment Infra (terraform)
+Create the necessary terraform code to build the AWS infrastructure needed for deploying into our `vagov-utility` VPC.
+
+:warning: There's a bidirectional dependency on the ansible deployment scripts & aws infra needed - sometimes you don't know which infrastructure is needed until you have the deployment config and sometimes you don't know which deployment configuration is needed until you know what infrastrucutre is available. :warning:
+
+
+#### Phase 3 - Observe, Modify, Iterate
+In this phase, we observe the performance of our Sentry 10 instances. If performance is not adequte, we can begin replacing some of the docker containers with dedicated AWS infrastructure. See #Future-Work
+
+
+### Alternatives
+In the first iteration of Sentry 10, the plan was to use multiple instances behind a load balancer. [You can see that design here](https://github.com/department-of-veterans-affairs/va.gov-team/blob/ccd761052fc3bc32f8c0c15689e2d7deccc900e0/teams/vsp/teams/tools/backend/sentry_design.md). However, there are [not enough available IP addresses in the `vagov-utility` VPC](https://github.com/department-of-veterans-affairs/va.gov-team/issues/1402) to house a load balancer.
+
+![image](https://user-images.githubusercontent.com/3077884/79232599-b861b900-7e35-11ea-8c5b-63b46f67cca4.png)
+
+As a result, the new design does away with the load balancer entirely in favor of a singular Sentry10 EC2 instance.
+
+
+### Future Work
+
+We'd like to tease out components from the `docker-compose` YAML file defined in the [`getsentry/onpremise`](https://github.com/getsentry/onpremise) bootstrap repo, and instead use AWS infrastructure.
```

Comment:
It'd be great to go into why you would want this and if you plan to work with upstream to get the work public.

Reply:
👍 , but who is "upstream" and what do you mean, "get the work public"?